searx/plugins/hostname_replace.py

```diff
@@ -0,0 +1,32 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+import re
+from urllib.parse import urlunparse
+from searx import settings
+from searx.plugins import logger
+from flask_babel import gettext
+
+name = gettext('Hostname replace')
```

Comment:
I guess translating the `name` is in conflict with the `enable_plugins` from settings.yml below / but see my comment below ..

Reply:
Good catch!

That's mean that this plugin and others can enable / disable according to the interface language --> it is another issue.

For reference:
https://github.com/searxng/searxng/blob/76e0f6807ccddb12ca9efe3b2343735d6e1777d4/searx/plugins/__init__.py#L81

https://github.com/searxng/searxng/blob/76e0f6807ccddb12ca9efe3b2343735d6e1777d4/searx/preferences.py#L307-L318

https://github.com/searxng/searxng/blob/76e0f6807ccddb12ca9efe3b2343735d6e1777d4/searx/plugins/oa_doi_rewrite.py#L9

https://github.com/searxng/searxng/blob/76e0f6807ccddb12ca9efe3b2343735d6e1777d4/searx/plugins/infinite_scroll.py#L3

https://github.com/searxng/searxng/blob/76e0f6807ccddb12ca9efe3b2343735d6e1777d4/searx/plugins/search_on_category_select.py#L18

https://github.com/searxng/searxng/blob/76e0f6807ccddb12ca9efe3b2343735d6e1777d4/searx/plugins/self_info.py#L19

https://github.com/searxng/searxng/blob/76e0f6807ccddb12ca9efe3b2343735d6e1777d4/searx/plugins/tracker_url_remover.py#L27

https://github.com/searxng/searxng/blob/76e0f6807ccddb12ca9efe3b2343735d6e1777d4/searx/plugins/vim_hotkeys.py#L3

Reply:
And actually it works:

when this line is executed outside a Flask request
https://github.com/searxng/searxng/blob/76e0f6807ccddb12ca9efe3b2343735d6e1777d4/searx/plugins/__init__.py#L81

so the locale is `en` :
https://github.com/searxng/searxng/blob/76e0f6807ccddb12ca9efe3b2343735d6e1777d4/searx/webapp.py#L238-L245

so `plugin.id` is set to the English label and not changed later.

Reply:
> And actually it works: when this line is executed outside a Flask request
> https://github.com/searxng/searxng/blob/76e0f6807ccddb12ca9efe3b2343735d6e1777d4/searx/plugins/__init__.py#L81
>

I'm not sure I understand ... Did you mean, it works because everything is done at import time ..

https://github.com/searxng/searxng/blob/2de95af92330c83c7b1aecc146e0f52e8825c59b/searx/plugins/__init__.py#L29-L36

and by luck this done before `flask_babel` is initialized and `flask.get_domain` returns a default `flask_babel.Domain()`

If so we should open another issue for this.

Reply:
> If so we should open another issue for this.

It is not by luck, but by design: see https://github.com/python-babel/flask-babel/blob/cc56bd9a4e7f614a4c1bf65c7f8b50b859359832/flask_babel/__init__.py#L671-L675

I've the feeling that the plugins should be referenced by their Python module name (in the settings, in the cookies)

Yes another issue should be opened for this.

I close the conversation here, since it's OT.